acceptancetests/jujupy/client.py

```diff
@@ -2121,6 +2128,90 @@ def switch(self, model=None, controller=None):
             raise ValueError('No target to switch to has been given.')
         self.juju('switch', (':'.join(args),), include_e=False)
 
+    def scp(self, _from, _to):
+        self.juju('scp', (_from, _to))
+
+
+# caas `add-k8s` did not implement parsing kube config path via flag,
+# so parse it via env var ->  https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/loader.go#L44
+KUBE_CONFIG_PATH_ENV_VAR = 'KUBECONFIG'
+
+
+class CaasClient(ModelClient):
+
+    cloud_name = 'k8cloud'
+    kubectl_path = None
+    kube_home = None
+    kube_config_path = None
+
+    def _ensure_dir(self, path):
+        try:
+            os.makedirs(path)
+        except OSError:
+            if not os.path.isdir(path):
+                raise
+
+    def __prepare(self):
+        # to pick up the new tmp juju_home, have to do this!
+        self.kubectl_path = os.path.join(self.env.juju_home, 'kubectl')
+        self.kube_home = os.path.join(self.env.juju_home, '.kube')
+        self.kube_config_path = os.path.join(self.kube_home, 'config')
+
+    def cluster_up(self, bundle_path):
+        self.__prepare()
+
+        self.deploy_bundle(bundle_path, static_bundle=True)
+        # Wait for the deployment to finish.
+        self.wait_for_started()
+
+        # ensure kube home
+        self._ensure_dir(self.kube_home)
```

Comment:
This should be part of __prepare (which should be in ```__init__```)

Reply:
the reason I put it here is the below two `scp` is the reason for ensuring `kube_home`, so I prefer to put related code together.